docker/Gemfile

```diff
@@ -1,7 +1,7 @@
 source ENV['GEM_SOURCE'] || "https://rubygems.org"
 
 def location_for(place)
-  if place =~ /^(git[:@][^#]*)#(.*)/
+  if place =~ /^((?:git|https?)[:@][^#]*)#(.*)/
```

Comment:
What is the first `?` doing in this?

Reply:
`(?:)` in a ruby regex means this is a non-capturing group, so `$1` will still be the whole url, and `$2` will still be the branch

Reply:
oooh, cool, thanks!